src/structs/vpx_bool_reader.rs

```diff
@@ -144,16 +144,18 @@ impl<R: Read> VPXBoolReader<R> {
 
         let probability = branch.get_probability() as u32;
 
-        let split = 1 + (((tmp_range - 1) * probability) >> BITS_IN_BYTE);
-        let big_split = (split as u64) << BITS_IN_LONG_MINUS_LAST_BYTE;
+        let big_split = ((tmp_range >> BITS_IN_BYTE) * probability
+            + ((256 - probability) << (BITS_IN_VALUE_MINUS_LAST_BYTE - BITS_IN_BYTE)))
+            & (255 << BITS_IN_VALUE_MINUS_LAST_BYTE);
+
```

Comment:
This is a bit confusing... maybe a comment describing what the original code did and how this improvement works?

Reply:
Yep, I'll add a comment here.

Reply:
Done.

Reply:
does this simplify to:

let split = ((((tmp_range - 0x1000000) >> 8) * probability) & 0xff000000) + 0x1000000;

Reply:
Yep, but this one is fully serial with dependence chain length 5, and 
``` rust
let big_split = ((tmp_range >> 8) * probability + ((256 - probability) << 16)) & 0xff000000;
```
has 4 as `(tmp_range >> 8) * probability` and `((256 - probability) << 16)` can be calculated in parallel.

Reply:
However, I see no difference in performance between these variants, so if you prefer this variant, I can use it.